spec/services/search/postgres/tag_spec.rb

```diff
@@ -0,0 +1,56 @@
+require "rails_helper"
+
+RSpec.describe Search::Postgres::Tag, type: :service do
+  describe "::search_documents" do
+    it "does not find non supported tags" do
+      tag = create(:tag, supported: false)
+
+      expect(described_class.search_documents(tag.name)).to be_empty
+    end
+
+    it "returns data in the expected format" do
+      tag = create(:tag, supported: true)
+
+      result = described_class.search_documents(tag.name)
+
+      expect(result.first.keys).to match_array(
+        %w[id name hotness_score rules_html supported short_summary],
+      )
+    end
+
+    it "finds a tag by its name" do
+      tag = create(:tag, supported: true)
+
+      expect(described_class.search_documents(tag.name)).to be_present
+    end
+
+    it "finds a tag by a partial name" do
+      tag = create(:tag, supported: true)
+
+      expect(described_class.search_documents(tag.name.first(1))).to be_present
+    end
+
+    it "finds multiple tags whose names have common parts", :aggregate_failures do
+      java = create(:tag, name: "java")
+      javascript = create(:tag, name: "javascript")
+      ruby = create(:tag, name: "ruby")
+
+      result = described_class.search_documents("jav")
+      tags = result.map { |r| r["name"] }
+
+      expect(tags).to include(java.name)
+      expect(tags).to include(javascript.name)
+      expect(tags).not_to include(ruby.name)
+    end
+
+    it "order tags by decreasing hotness score" do
+      java = create(:tag, name: "java", hotness_score: 10)
+      javascript = create(:tag, name: "javascript", hotness_score: 20)
+
+      result = described_class.search_documents("jav")
+      tags = result.map { |r| r["name"] }
+
+      expect(tags).to match_array([javascript.name, java.name])
```

Comment:
`match_array` is an alternate form of `contain_exactly`, which does not take ordering into account.

[`match_array` docs](https://www.rubydoc.info/gems/rspec-expectations/RSpec%2FMatchers:match_array):

> An alternate form of `contain_exactly `that accepts the expected contents as a single array arg rather that splatted out as individual items.

```ruby
expect(results).to contain_exactly(1, 2)
# is identical to:
expect(results).to match_array([1, 2])
```

[`contain_exactly` docs](https://www.rubydoc.info/gems/rspec-expectations/RSpec/Matchers#contain_exactly-instance_method), emphasis added:

> Passes if actual contains all of the expected **regardless of order**.

I threw together a [test spec](https://gist.github.com/citizen428/725ea91032f7c270158afaf667d573d3) to illustrate this.

Since this spec is specifically about ordering we should use a different matcher instead;

```suggestion
      expect(tags).to eq([javascript.name, java.name])
```

Reply:
This stuff is why I dislike RSpec :D